protocol/oracle-manager/contracts/nodes/UniswapNode.sol

```diff
@@ -104,11 +104,25 @@ library UniswapNode {
                 (address, address, uint8, uint8, address, uint32)
             );
 
-        if (IERC20(token0).decimals() != decimals0) {
+        if (IERC20(token).decimals() != decimals0) {
+            return false;
+        }
+
+        if (IERC20(stablecoin).decimals() != decimals1) {
+            return false;
+        }
+
+        address poolToken0 = IUniswapV3Pool(pool).token0();
+        address poolToken1 = IUniswapV3Pool(pool).token1();
+
+        if (
+            !(poolToken0 == token && poolToken1 == stablecoin) &&
+            !(poolToken0 == stablecoin && poolToken1 == token)
+        ) {
             return false;
         }
 
-        if (IERC20(token1).decimals() != decimals1) {
+        if (decimals0 > 18 || decimals1 > 18) {
```

Comment:
this could cause problems in the future for supporting tokens that we want to support. we have a function in v3 which effectively handles the issue of what happens if a token has > 18 decimals. are we sure this is the path we want to take (at least for right now?)